cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -220,18 +220,70 @@ pub enum Inst {
         srcloc: Option<SourceLoc>,
     },
 
-    /// XMM (scalar) unary op (from xmm to integer reg): movd, movq
+    /// XMM (scalar) unary op (from xmm to integer reg): movd, movq, cvtts{s,d}2si
     XmmToGpr {
         op: SseOpcode,
         src: Reg,
         dst: Writable<Reg>,
+        dst_size: OperandSize,
     },
 
-    /// XMM (scalar) unary op (from integer to float reg): movd, movq
+    /// XMM (scalar) unary op (from integer to float reg): movd, movq, cvtsi2s{s,d}
     GprToXmm {
         op: SseOpcode,
         src: RegMem,
         dst: Writable<Reg>,
+        src_size: OperandSize,
+    },
+
+    /// Converts an unsigned int64 to a float64.
+    CvtUint64ToFloatSeq {
+        /// Is the target a 64-bits or 32-bits register?
+        to_f64: bool,
+        src: Reg,
+        dst: Writable<Reg>,
+        tmp_gpr1: Writable<Reg>,
+        tmp_gpr2: Writable<Reg>,
+    },
+
+    /// Converts a scalar xmm to a signed int32/int64.
+    CvtFloatToSintSeq {
```

Comment:
It seems like the two comments are inconsistent:
```
/// Converts an unsigned int64 to a float64.
/// Is the target a 64-bits or 32-bits register?
```

Reply:
@julian-seward1 Indeed, fixed the above comment for `CvtUint64ToFloatSeq` so it says `to a float32/float64`, thanks!

@cfallin I had thought of this; it just makes the emit code very long, and it was nice to have one axis to cut them. I hesitated between signed vs unsigned, or saturated vs not saturated, but the sat instructions emit very similar code, so splitting signed and unsigned made more sense. We could merge them back at some point, maybe using an enum to avoid manipulating the two booleans.